conans/util/files.py

```diff
@@ -212,6 +212,21 @@ def rmdir(path):
         raise
 
 
+def _change_file_permissions(func, path, exc_info):
+    os.chmod(path, stat.S_IRWXU)
+    os.remove(path)
+
+
+def remove(path):
+    try:
+        assert os.path.isfile(path)
+        shutil.rmtree(path, onerror=_change_file_permissions)
```

Comment:
I don't understand why this work, `shutil.rmtree` usually fails if you don't pass a file, is the "onerror" making it happen somehow?

Reply:
That's it, it fails and the `onerror` function is handling it (it was taken from an example where the delete file/dir was handled by the same function).

I will change it to try-os.remove-catch and handle the permission error, it is better is this function is only going to accept a file 👍